### PR TITLE
NAV-57 Add action checklist endpoint

### PR DESCRIPTION
**Pull request by jonathansberry**
This PR adds an action checklist endpoint. 

At this time the checklist only covers actions in the action breadcrumbs from calling decide.  Further work may be done later to forecast future actions that will need to be completed.  

See line 187+ of `test_endpoints.py` for an example response showing how I'm proposing the checklist is is structured. 

We return the overall progress and a "checklist", which is a list of actions and milestones, where milestones are distinguished from actions because they have their own sub checklist. 

For each action I am returning "skipped", "reached" (is there a better name? I'm meaning to indicate whether or not the action is a future action later down the graph) and "complete" being defined as "not skipped or not reached".

### CODE REVIEW

File: navigator_engine/api.py [hunk 1]
```diff
@@ -20,37 +20,19 @@ def decide():
...
-        data_loader=data_loader,
-        skip_requests=skip_requests,
-        stop=stop_action
-    )
+    engine = create_engine()
     engine.decide()
     del engine.decision['node']
 
-    if stop_action and stop_action != engine.decision['id']:
+    if engine.stop_action and engine.stop_action != engine.decision['id']:
```
Comment (tomeksabala) [inline navigator_engine/api.py:32]: That's a little misleading. We're including the current decision id in the breadcrumbs. So calling decide with `stopAction = breadcrumbs[-1]['id]` should return a valid decision IMO. What do you think?
I have send exactly this request yesterday and after getting error "The actionID is not in the action path" I thought it's an engine bug.

**Review by tomeksabala — COMMENTED** [review]
I have one comment which I'd like to talk about before getting this merged in.

**Review by tomeksabala — APPROVED** [review]

File: navigator_engine/tests/integration_tests/test_endpoints.py [hunk 2]
```diff
@@ -183,3 +183,77 @@ def setup_endpoint_test(mocker, data=None):
...
+        'data': {
+            'url': 'https://example.ckan/api/3/action/package_show?id=example',
+            'authorization_header': "example-api-key"
+        },
+        'skipActions': ['tst-1-5-a', 'tst-1-6-a']
+    }))
+    assert response.json == {
+        'progress': 100,
+        'checklist': [{
+            'complete': True,
```
Comment (tomeksabala) [inline navigator_engine/tests/integration_tests/test_endpoints.py:206]: shoudn't "complete" be renamed to "terminus"?

**Review by tomeksabala — CHANGES_REQUESTED** [review]
It was agreed to change the edpoint to `/api/action/list`
Example response:
```
{
  "milestones": [
    {"id", "title", "progress"}
  ],
  "actions": [
    {"id", "milestoneID":  <nullable>, "title", "skipped", "reached", "manualConfirmationRequired"}

  ]
}
```

**Comment by jonathansberry** [comment]
This is replacef by https://github.com/fjelltopp/navigator_engine/pull/55

**Pull request closed without merging**